### PR TITLE
Database

### LINKED ISSUE #2: Implement DB

### PR DESCRIPTION
**Pull request by dudebehinddude**
Implements a database into the bot
Currently, the database is used to store whether or not the user has agreed to the terms of using DND commands, and prevents them from using them unless they have agreed to the terms.
Closes #2

### CODE REVIEW

File: Bot/Commands/DND/RaceInfo.cs [hunk 1]
```diff
@@ -20,6 +21,9 @@ public class RaceInfo : ISlashCommand
 
         public async Task CommandEvent(SocketSlashCommand slashCommand)
         {
+            bool tos = await CheckDNDTos(slashCommand);
```
Comment (035WasTaken) [inline Bot/Commands/DND/RaceInfo.cs:24]: This makes the code repetitive if we have to include this line in the CommandEvent() method of every command we make, move it to the interaction handler, you still have access to the SocketSlashCommand there

File: Bot/Services/ComponentHandler.cs [hunk 2]
```diff
@@ -85,5 +87,40 @@ public async Task RaceMenuInteractionHandler(SocketMessageComponent component)
 
             await component.Channel.SendMessageAsync(embed: embed.Build());
         }
+
+        public static async Task TermsInteractionHandler(SocketMessageComponent component)
+        {
+            bool acceptTerms;
+            switch (component.Data.CustomId)
```
Comment (035WasTaken) [inline Bot/Services/ComponentHandler.cs:94]: Use a switch expression here, it's a bit more intuitive

File: Bot/Services/DatabaseHandler.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    OracleCommand createUser = new("INSERT INTO users (id) VALUES (:userID)", connection);
+                    createUser.Parameters.Add("userID", (long)id);
+                    await createUser.ExecuteNonQueryAsync();
+                }
+            }
+        }
+
+        public static async Task<(bool? previous, bool? result)> UpdateDNDTerms(ulong discordId, bool value)
+        {
+            int agreeToTerms = value ? 1 : 0;
```
Comment (035WasTaken) [inline Bot/Services/DatabaseHandler.cs:77]: What's the point in using an int as opposed to a bool for the db ? The performance save is negligible and this way I feel can just cause further confusion down the line when some things are `true` or `false` and others are `1` or `0`

File: Bot/Services/DatabaseHandler.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    dbValue = (short?)await checkValue.ExecuteScalarAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                log.Error(ex);
+                return (null, null);
+            }
+
+            bool previous = (pdbValue == 0) ? false : true;
```
Comment (035WasTaken) [inline Bot/Services/DatabaseHandler.cs:108]: If the value was `bool` to begin with, ternary wouldn't be required here. This method reduces readability.

File: Bot/Commands/DND/AgreeToTerms.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        public SlashCommandBuilder SlashCommandBuilder => new SlashCommandBuilder()
+            .WithName("agreetoterms")
+            .WithDescription("List the terms associated with using DND related commands, and choose to either accept or deny them");
+
+        public ulong? Guild => null;
+
+        public async Task CommandEvent(SocketSlashCommand command)
+        {
+            EmbedBuilder embed = new EmbedBuilder()
```
Comment (035WasTaken) [inline Bot/Commands/DND/AgreeToTerms.cs:22]: This part isn't really related to this PR specifically, but in the future we should try to handle this part outside of the command classes

**Review by 035WasTaken — CHANGES_REQUESTED** [review]
There are a few small things I don't like which I've pointed out, but for the most part it's good. Change those and I think this is fine.

File: Bot/Services/DatabaseHandler.cs (hunk 4 above)
Comment (035WasTaken) [reply Bot/Services/DatabaseHandler.cs:108]: Nevermind, it turns out Oracle just sucks.

Comment (DUDEbehindDUDE) [reply Bot/Services/DatabaseHandler.cs:108]: yeah lol

**Comment by DUDEbehindDUDE** [comment]
^ what happens when you amend a commit you've already pushed

**Pull request merged**